src/utils/havoc.js

```diff
@@ -446,8 +446,34 @@ class ObjectValueHavocingVisitor {
   }
 
   visitAbstractValue(val: AbstractValue): void {
-    for (let i = 0, n = val.args.length; i < n; i++) {
-      this.visitValue(val.args[i]);
+    if (!val.mightBeObject()) {
+      // Only objects need to be havoced.
+      return;
+    }
+    if (val.values.isTop()) {
+      // If we don't know which object instances it might be,
+      // then it might be one of the arguments that created
+      // this value. See #2179.
+
+      // To ensure that we don't forget to provide arguments
+      // that can be havoced, we require at least one argument.
+      let whitelistedKind =
+        val.kind &&
+        (val.kind === "widened numeric property" || // TODO: Widened properties needs to be havocable.
+          val.kind.startsWith("abstractCounted"));
```

Comment:
It took me a while to convince myself that this is OK. Essentially, anything created this way is "pre-havoced", i.e. we don't know the value of a property of such an object and we get a new one every time we access the property. At the very least we should provide and efficient and much more obviously correct way of checking that an abstract value is of this kind.

Reply:
On second thoughts: We can only access properties of abstract objects with templates and such object will have a values domain that is not top.

Reply:
Optimized functions get these arguments passed into them. https://github.com/facebook/prepack/blob/3deb5e2885a689e7b8be5bb8ad88e24fe8e4fecf/src/serializer/functions.js#L150

These two are the only cases of empty args that occurs in test coverage.

It is ok not to havoc them since they're arguments to pure functions, they can't be mutated.